doc/errors.org

```diff
@@ -0,0 +1,95 @@
+* Errors
+https://ziglang.org/documentation/0.10.1/#Errors
+In ZIG errors are juste values.
+Those values are those of an special type of enum "error".
+When you declare your enum using this keyword instead of "enum", the compiler is going to know that those values are going to be errors, therefore they can be catched, tried, ...
+
+
+** Handling errors in the application flow
+Functions can return either a value or an error, this is done by adding the "!" to the return type of a function
+With the "!" the return type of the function is the coerced "anyerror!u64"
```

Comment:
In the 0.11 zig doc, i found that error are coerced to `u16`, is this specific ?

Reply:
https://ziglang.org/documentation/0.11.0/#Errors
You are probably reffering to this: "The number of unique error values across the entire compilation should determine the size of the error set type. However right now it is hard coded to be a u16. See [#768](https://github.com/ziglang/zig/issues/786)."

What I meant with this is that when we add the "!" my return value becomes something like "anyerror!string", "anyerror!bool".

And yes what you said is correct from my understanding "anyerror" is simply an "u16" under the hood.

My documentation was indeed not precise, not in relation with my exemple and could lead to misenterpretation, I changed it ty :)

`With the "!" the return type of the function below is the coerced "anyerror!u8"

When a function can return an error we have to either "try" or "catch" it.

#+source: simple
#+begin_src zig :imports '(std)
fn maybeErrorMaybeValue(isError: bool) !u8 {
    // ... might return an u8 or an error
    if (isError) {
        return error.Bar;
    } else {
        return 2;
    }
}`

I don't know if saying in my doc what is "anyerror" under the hood is  useful, what do you think ?

Reply:
Ok, I understand. I think that it's not necessary to include the information about the issue [#786](https://github.com/ziglang/zig/issues/786).
But we can say it more explicitly as in the standard doc [chapter : error-union-type](https://ziglang.org/documentation/0.11.0/#Error-Union-Type)